test/e2e/e2e.go

```diff
@@ -294,20 +297,28 @@ func createPod(f *framework.Framework, podName, nodeSelector, namespace string,
 			RestartPolicy: v1.RestartPolicyNever,
 		},
 	}
+
+	for _, o := range options {
+		o(pod)
+	}
+
 	podClient := f.ClientSet.CoreV1().Pods(namespace)
-	_, err := podClient.Create(context.Background(), pod, metav1.CreateOptions{})
+	res, err := podClient.Create(context.Background(), pod, metav1.CreateOptions{})
 	if err != nil {
 		framework.Logf("Warning: Failed to get logs from pod %q: %v", pod.Name, err)
 	}
-	err = e2epod.WaitForPodNotPending(f.ClientSet, podName, namespace)
-	if err != nil {
-		logs, logErr := e2epod.GetPodLogs(f.ClientSet, namespace, pod.Name, contName)
-		if logErr != nil {
-			framework.Logf("Warning: Failed to get logs from pod %q: %v", pod.Name, logErr)
-		} else {
-			framework.Logf("pod %s/%s logs:\n%s", namespace, pod.Name, logs)
-		}
-	}
+	err = e2epod.WaitForPodNotPending(f.ClientSet, namespace, podName)
+	framework.ExpectNoError(err)
+
+	logs := ""
+	gomega.Eventually(func() error {
+		var err error
+		logs, err = e2epod.GetPodLogs(f.ClientSet, namespace, pod.Name, contName)
+		return err
+	}, 30*time.Second, 1*time.Second).Should(gomega.BeNil(), "Failed to get logs from pod", pod.Name)
+
+	framework.Logf("pod %s/%s logs:\n%s", namespace, pod.Name, logs)
```

Comment:
I think this isn't going to do what you want - we want to get pod logs in all cases when the pod doesn't stay in Running state. But the assertin on line 311 will fail first.

Reply:
Ah! I totally mis-interpreted it. Will change, thanks!

Reply:
and now, I think I remember why I added the active waiting.
It was an attempt before realizing that the order of parameters we were calling WaitForPodNotPending was wrong